### PR TITLE
#167727724 allow a user to create an account

### PR DESCRIPTION
**Pull request by ladaposamuel**
#### What does this PR do?
This PR will enable user account creation

#### Description of Task to be completed?
- create sign up controller
- create user model
- create user migration file
- create user seeder
- create user sign up route

#### How should this be manually tested?
- CLONE this repo
- open the cloned directory
- run `npm install`
- run `npm run start:dev`
- access the route using `api/v1/user/signup`

#### What are the relevant pivotal tracker stories?

[#167727724](https://www.pivotaltracker.com/story/show/167727724)

### CODE REVIEW

File: src/config/swaggerDoc.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ const swaggerDefinition = {
 
 const options = {
   swaggerDefinition,
-  apis: ['src/routes/*.js'],
+  apis: ['src/routes/api/*.js'],
```
Comment (ladaposamuel) [inline src/config/swaggerDoc.js]: Try adding a second value to generate the docs for the index routes

`src/routes/*js`

File: src/controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const {
+    firstName, lastName, email, password, phoneNo
+  } = req.body;
+  try {
+    const user = {
+      firstName,
+      lastName,
+      email,
+      phoneNo,
+      password: authHelper.hashedPassword(password)
```
Comment (ladaposamuel) [inline src/controllers/userController.js]: Change line, enable sequelize hooks.

File: src/controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    };
+    const createdUser = await User.create(user);
```
Comment (ladaposamuel) [inline src/controllers/userController.js]: sync with @Mcdavid95 and use a user service

File: src/controllers/userController.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    const userData = {
+      user: {
+        id: createdUser.id,
+        email: createdUser.email,
+        token: authHelper.generateToken({ id: createdUser.id }),
+        bio: createdUser.bio,
```
Comment (ladaposamuel) [inline src/controllers/userController.js]: remove

File: src/controllers/userController.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        image: createdUser.avatar
```
Comment (ladaposamuel) [inline src/controllers/userController.js]: remove

File: src/controllers/userController.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    };
+
+    return response(res, 200, 'success', userData);
+  } catch (e) {
+    return response(res, 500, 'false', {
```
Comment (ladaposamuel) [inline src/controllers/userController.js]: change the false status message to error

File: src/database/seeders/20190819153902-user.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+const bcrypt = require("bcrypt");
```
Comment (ladaposamuel) [inline src/database/seeders/20190819153902-user.js]: remove once hooks are set

File: src/test/routes/auth.spec.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+
+import chai from 'chai';
```
Comment (ladaposamuel) [inline src/test/routes/auth.spec.js]:
```
import {
  app, messages, chai, expect
} from '../testHelpers/config';

```

change

File: package.json [hunk 9]
```diff
@@ -4,6 +4,7 @@
   "description": "An app for making company global travel and accommodation easy and convenient",
   "main": "index.js",
   "scripts": {
+    "db-setup": "NODE_ENV=test sequelize db:migrate:undo:all && NODE_ENV=test sequelize db:migrate && npm run seed",
```
Comment (ladaposamuel) [inline package.json]: change to db:setup

File: src/utils/authHelper.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcrypt';
+import { config } from 'dotenv';
```
Comment (ladaposamuel) [inline src/utils/authHelper.js]: Try `import './config/env';` to import the env configs

File: src/utils/authHelper.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+config();
+
+const secret = process.env.SECRET_KEY;
+const saltRounds = 7;
```
Comment (ladaposamuel) [inline src/utils/authHelper.js]: change saltrounds value to get from env

File: src/database/models/user.js [hunk 12]
```diff
@@ -1,14 +1,48 @@
-module.exports = (sequelize, DataTypes) => {
-  const User = sequelize.define('User', {
-    first_name: {
-      type:  DataTypes.STRING,
+import bcrypt from 'bcrypt';
+
+const salt = bcrypt.genSaltSync(parseInt(process.env.SALT_ROUNDS));
+parseInt
```
Comment (tejiri4) [inline src/database/models/user.js]: Kindly remove the parseInt here

File: .env-sample [hunk 13]
```diff
@@ -1,2 +1,4 @@
 DATABASE_URL="postgres://username:password@localhost:5432/database_name"
 TEST_DATABASE_URL="postgres://username:password@localhost:5432/database_name"
+SECRET_KEY=""
+SALT_ROUNDS=
```
Comment (tejiri4) [inline .env-sample]: add new line

File: src/database/seeders/20190819153902-user.js [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+        phoneNo: '2347032123404',
+        createdAt: new Date(),
+        updatedAt: new Date()
+      }
+    ], {});
+  },
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.bulkDelete('Users', null, {});
+  }
+};
```
Comment (tejiri4) [inline src/database/seeders/20190819153902-user.js]: add new line

File: src/database/models/user.js (hunk 12 above)
Comment (tejiri4) [reply src/database/models/user.js]: @ladaposamuel

Comment (ladaposamuel) [reply src/database/models/user.js]: it wont work without it, the error massage was that salt rounds should be a number

Comment (tejiri4) [reply src/database/models/user.js]: Line 4 I mean I removed it and it works @ladaposamuel

**Comment by tejiri4** [comment]
Kindly rebase with develop so I merge your work

**Review by tejiri4 — APPROVED** [review]
LGTM

**Pull request merged**